lib/connectors/mongodb/connector.rb

```diff
@@ -56,30 +70,166 @@ def initialize(configuration: {})
+
+        function, args = setup_db_function_with_args(advanced_filter_config)
+
+        call_db_function_on_collection(function, args, &on_doc_serialization)
+      end
+
+      private
+
+      def extract_rules(job_description)
+        Utility::Commons.return_if_present(job_description.dig(:filtering, :rules), DEFAULT_RULES)
+      end
+
+      def extract_filter_config(job_description)
+        Utility::Commons.return_if_present(job_description.dig(:filtering, :advanced_config), DEFAULT_ADVANCED_CONFIG)
+      end
+
+      def setup_db_function_with_args(filter_config)
+        # default parameters
+        function = FIND
+        args = NO_ARGS
+
+        function, args = setup_find(filter_config) if find_present?(filter_config)
+
+        function, args = setup_aggregate(filter_config) if aggregate_present?(filter_config)
+
+        [function, args]
+      end
+
+      def check_filter_config!(rules, config)
+        return unless filtering_present?(rules, config)
+
+        check_find_and_aggregate_present!(config)
+
+        check_find_and_aggregate_missing!(config)
+      end
+
+      def check_find_and_aggregate_missing!(config)
+        allowed_keys = Utility::Strings.format_string_array(ALLOWED_TOP_LEVEL_FILTER_KEYS, default: ' ')
+        keys_present = Utility::Strings.format_string_array(config.keys, default: ' ')
+
+        find_and_aggregate_missing = !find_present?(config) && !aggregate_present?(config)
+        both_missing_msg = "Only one of #{allowed_keys} is allowed in the filtering object. Keys present: '#{keys_present}'."
+
+        raise Utility::InvalidFilterConfigError.new(both_missing_msg) if find_and_aggregate_missing
+      end
+
+      def check_find_and_aggregate_present!(config)
+        find_and_aggregate_present = find_present?(config) && aggregate_present?(config)
+        both_present_msg = '\'find\' and \'aggregate\' functions cannot be used at the same time for MongoDB. Please drop one from the configuration.'
+
+        raise Utility::InvalidFilterConfigError.new(both_present_msg) if find_and_aggregate_present
+      end
+
+      def call_db_function_on_collection(function, args, &on_doc_serialization)
         with_client do |client|
           # We do paging using skip().limit() here to make Ruby recycle the memory for each page pulled from the server after it's not needed any more.
           # This gives us more control on the usage of the memory (we can adjust PAGE_SIZE constant for that to decrease max memory consumption).
           # It's done due to the fact that usage of .find.each leads to memory leaks or overuse of memory - the whole result set seems to stay in memory
           # during the sync. Sometimes (not 100% sure) it even leads to a real leak, when the memory for these objects is never recycled.
-          cursor = client[@collection].find
+          cursor = client[@collection].send(function, *args)
           skip = 0
 
+          options = args[1]
```

Comment:
Options are the same for retrieval functions (find/aggregate) in MongoDB (also at the same place), therefore we can assume that they could exist as second parameter.